client.go

```diff
@@ -170,7 +170,7 @@ func (c *Client) request(options *RequestOptions) (*Response, error) {
 
 	err = json.Unmarshal(data, resp)
 	if err != nil {
-		return resp, err
+		return resp, newRemoteError(0, err.Error())
```

Comment:
I'm not sure this should be a remote error?  It happens when unmarshalling the json, after the remote request is done.  I recognize the content from the remote request is supposed to be json though, so I'm persuadable.  In that case I don't think the code should be 0.  At the very least a 50x error I would think.